resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```
@@ -673,23 +676,28 @@ private[spark] class Client(
         .build())
   }
 
-  private def buildSubmissionRequest(): KubernetesCreateSubmissionRequest = {
-    val appResourceUri = Utils.resolveURI(mainAppResource)
-    val resolvedAppResource: AppResource = appResourceUri.getScheme match {
-      case "file" | null =>
-        val appFile = new File(appResourceUri.getPath)
+  private def buildSubmissionRequest(
+      submitterLocalFiles: Iterable[String],
+      submitterLocalJars: Iterable[String]): KubernetesCreateSubmissionRequest = {
+    val mainResourceUri = Utils.resolveURI(mainAppResource)
+    val resolvedAppResource: AppResource = Option(mainResourceUri.getScheme)
+        .getOrElse("file") match {
+      case "file" =>
+        val appFile = new File(mainResourceUri.getPath)
         if (!appFile.isFile) {
           throw new IllegalStateException("Provided local file path does not exist" +
```

this check is now redundant with the check in line 87?  remove this one since the other one is earlier and failing fast is good